lua/nvim-treesitter-refactor/smart_rename.lua

```diff
@@ -10,39 +10,46 @@ local api = vim.api
 local M = {}
 
 function M.smart_rename(bufnr)
-  local bufnr = bufnr or api.nvim_get_current_buf()
+  bufnr = bufnr or api.nvim_get_current_buf()
   local node_at_point = ts_utils.get_node_at_cursor()
+  local function complete_rename(new_name)
+    -- Empty name cancels the interaction or ESC
+    if not new_name or #new_name < 1 then
+      return
+    end
 
-  if not node_at_point then
-    utils.print_warning "No node to rename!"
-    return
-  end
+    local definition, scope = locals.find_definition(node_at_point, bufnr)
+    local nodes_to_rename = {}
+    nodes_to_rename[node_at_point:id()] = node_at_point
+    nodes_to_rename[definition:id()] = definition
 
-  local node_text = ts_utils.get_node_text(node_at_point)[1]
-  local new_name = vim.fn.input("New name: ", node_text or "")
+    for _, n in ipairs(locals.find_usages(definition, scope, bufnr)) do
+      nodes_to_rename[n:id()] = n
+    end
 
-  -- Empty name cancels the interaction or ESC
-  if not new_name or #new_name < 1 then
-    return
-  end
+    local edits = {}
 
-  local definition, scope = locals.find_definition(node_at_point, bufnr)
-  local nodes_to_rename = {}
-  nodes_to_rename[node_at_point:id()] = node_at_point
-  nodes_to_rename[definition:id()] = definition
-
-  for _, n in ipairs(locals.find_usages(definition, scope, bufnr)) do
-    nodes_to_rename[n:id()] = n
+    for _, node in pairs(nodes_to_rename) do
+      local lsp_range = ts_utils.node_to_lsp_range(node)
+      local text_edit = { range = lsp_range, newText = new_name }
+      table.insert(edits, text_edit)
+    end
+    vim.lsp.util.apply_text_edits(edits, bufnr)
   end
 
-  local edits = {}
+  if not node_at_point then
+    utils.print_warning "No node to rename!"
+    return
+  end
 
-  for _, node in pairs(nodes_to_rename) do
-    local lsp_range = ts_utils.node_to_lsp_range(node)
-    local text_edit = { range = lsp_range, newText = new_name }
-    table.insert(edits, text_edit)
+  local node_text = ts_utils.get_node_text(node_at_point)[1]
+  local input = { prompt = "New name: ", default = node_text or "" }
+  if not vim.ui.prompt then
+    local new_name = vim.fn.input(input.prompt, input.default)
+    complete_rename(new_name)
+  else
+    vim.ui.prompt(input, complete_rename)
```

Comment:
Shouldn't this be `vim.ui.input`?

Reply:
You're right, thats weird, I don't know how that happened, must've been a stray search and replace